articles/android-nested-recycler-view/index.md

```diff
@@ -0,0 +1,400 @@
+        <ImageView
+            android:id="@+id/hero_image"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            android:src="@drawable/ic_launcher_background"
+            app:layout_constraintTop_toTopOf="parent" />
+    </androidx.cardview.widget.CardView>
+
+    <TextView
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:text="MovieTitle"
+        android:textColor="#C4DFEF"
+        android:textSize="12sp"
+        android:id="@+id/movie_name"
+        android:layout_marginTop="5dp"
+        android:layout_gravity="center" />
+</LinearLayout>
+```
+
+### Setting up Model classes
+
+These are java classes that utilize the necessary parameters for each in the RecyclerView. It consists of getters and setters method declaration. A model class contains a collection of data/objects that we want to show in the RecyclerView.
+
+The application will have two model classes.
+
+- Model one: contains information about the child items.
+
+This data class stores the Movie image and its name. The model contains one text view for the title and one image as the movie hero image.
+
+```java
+
+public class ChildModel {
+    private  int hero_image;
+    private String movieName;
+
+    public ChildModel(int hero_image, String movieName){
+        this.hero_image = hero_image;
+        this.movieName = movieName;
+    }
+    public int getHeroImage() {
+        return hero_image;
+    }
+    public String getMovieName() {
+        return movieName;
+    }
+}
+```
+
+- Model two: contains information about the parent items.
+
+This is a text view for the movie category. The parent item also consists of an array list of the child items. I will show you to bind them using a RecyclerView adapter holder.
+
+```java
+public class ParentModel {
+    private String movieCategory;
+
+    public ParentModel(String movieCategory) {
+        this.movieCategory = movieCategory;
+    }
+    public String movieCategory() {
+        return movieCategory;
+}
+```
+
+### Setting up Adapter classes
+
+Adapters are used to set and pass data that will be displayed in the respective RecyclerView items. The adapter manages the collection of items (defined in an individual layout) and `ViewHolder` objects. `ViewHolder` defines the individual views (widgets) in the layout and populate the components into an individual row layout. An `onBindViewHolder` class will bind the views with their respective data objects.
+
+Technically, the data could come from a database or a remote server if the information to be passed to the recycling view is hosted remotely. To keep this app straight to the point, we will use dummy/locally available information and then pass them to the respective RecyclerView Adapters.
+
+Let populate this data into the respective views. We will have two view adapters for each of the RecyclerView widgets. Create the two adapters.
```

Comment:
Consider changing "Let populate this data into the respective views." to "Let us populate this data into the respective views."